bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/internal/model/commands/GetVersionCommand.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands;
+
+import java.util.concurrent.Executor;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author blafois
+ *
+ */
+public class GetVersionCommand extends BRC1HCommand {
+
+    private final Logger logger = LoggerFactory.getLogger(GetVersionCommand.class);
+    private String remoteControllerVersion;
+    private String communicationControllerVersion;
+
+    @Override
+    public byte[] getRequest() {
+        return MadokaMessage.createRequest(this);
+    }
+
+    @Override
+    public boolean handleResponse(Executor executor, ResponseListener listener, byte @Nullable [] response) {
+        if (response == null) {
+            return false;
+        }
+
+        try {
+
+            MadokaMessage mm = MadokaMessage.parse(response);
+
+            int remoteControllerMajor = Integer.valueOf(mm.getValues().get(0x45).getRawValue()[0]);
+            int remoteControllerMinor = Integer.valueOf(mm.getValues().get(0x45).getRawValue()[1]);
+            int remoteControllerRevision = Integer.valueOf(mm.getValues().get(0x45).getRawValue()[2]);
+
+            this.remoteControllerVersion = remoteControllerMajor + "." + remoteControllerMinor + "."
+                    + remoteControllerRevision;
+
+            int commControllerMajor = Integer.valueOf(mm.getValues().get(0x46).getRawValue()[0]);
+            int commControllerMinor = Integer.valueOf(mm.getValues().get(0x46).getRawValue()[1]);
+
+            this.communicationControllerVersion = commControllerMajor + "." + commControllerMinor;
+
+            listener.receivedResponse(this);
+            setState(State.SUCCEEDED);
+            return true;
+        } catch (Exception e) {
+            logger.error("Error while parsing response", e);
```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. An error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

My general rule for logging levels is:
* trace - used for normal expected execution paths
* debug - used for unexepected execution paths, but well within normal operation
* info - used for notable points in an execution path, like a milestone. (in openhab we try reserve this logging level for the core, so bindings should rarely ever call this.)
* warn - used for notable unexpected execution paths that a regular user (not just a developer) should be notified of. Warnings should be used to indicate that something not-normal occurred and user intervention is required to resolve. Warnings do not indicate a failure to operate merely an abnormal condition of operation that can still be handled by the binding. Failures in binding operation should be indicated by changing the thing status to offline.
* error - used to indicate catastrophic program failure. This should be used to indicate a catastrophic failure in openhab's ability to operate. A failure in a binding would never cause openhab as a whole to fail so a failure in a bindings should never log an error. Instead that failure should be indicated by changing the thing status.

Please make changes elsewhere as appropriate.